### ISSUE TITLE
MultiScaleDeformableAttention module undefined symbol: _ZN3c104cuda20getCurrentCUDAStreamEa

### ISSUE THREAD

**Issue by BingfengHan** [fundamentalvision/Deformable-DETR#123]
Did someone meet and solve this problem?

**Comment by BingfengHan** [comment]
`bash make.sh `

```
which: no hipcc in (/home/...)
/home/.../python3.7/site-packages/setuptools/dist.py:726: UserWarning: Usage of dash-separated 'index-url' will not be supported in future versions. Please use the underscore name 'index_url' instead
  % (opt, underscore_opt)
running build
running build_py
running build_ext
running install
/home/.../python3.7/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  setuptools.SetuptoolsDeprecationWarning,
/home/.../python3.7/site-packages/setuptools/command/easy_install.py:159: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  EasyInstallDeprecationWarning,
running bdist_egg
running egg_info
writing manifest file 'MultiScaleDeformableAttention.egg-info/SOURCES.txt'
running install_lib
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/functions
creating build/bdist.linux-x86_64/egg/modules
byte-compiling build/bdist.linux-x86_64/egg/functions/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.linux-x86_64/egg/functions/ms_deform_attn_func.py to ms_deform_attn_func.cpython-37.pyc
byte-compiling build/bdist.linux-x86_64/egg/modules/ms_deform_attn.py to ms_deform_attn.cpython-37.pyc
byte-compiling build/bdist.linux-x86_64/egg/modules/__init__.py to __init__.cpython-37.pyc
byte-compiling build/bdist.linux-x86_64/egg/MultiScaleDeformableAttention.py to MultiScaleDeformableAttention.cpython-37.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying MultiScaleDeformableAttention.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MultiScaleDeformableAttention.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MultiScaleDeformableAttention.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying MultiScaleDeformableAttention.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
__pycache__.MultiScaleDeformableAttention.cpython-37: module references __file__
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
creating /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg
Extracting MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg to /home/.../python3.7/site-packages
byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/MultiScaleDeformableAttention.py to MultiScaleDeformableAttention.cpython-37.pyc
byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/functions/__init__.py to __init__.cpython-37.pyc
byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/functions/ms_deform_attn_func.py to ms_deform_attn_func.cpython-37.pyc
byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/modules/__init__.py to __init__.cpython-37.pyc
byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/modules/ms_deform_attn.py to ms_deform_attn.cpython-37.pyc
Adding MultiScaleDeformableAttention 1.0 to easy-install.pth file

Installed /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg
Processing dependencies for MultiScaleDeformableAttention==1.0
Finished processing dependencies for MultiScaleDeformableAttention==1.0
```

Did this installation correctly finish?

**Comment by Wenzhiqiang16** [comment]
> `bash make.sh `
> 
> ```
> which: no hipcc in (/home/...)
> /home/.../python3.7/site-packages/setuptools/dist.py:726: UserWarning: Usage of dash-separated 'index-url' will not be supported in future versions. Please use the underscore name 'index_url' instead
>   % (opt, underscore_opt)
> running build
> running build_py
> running build_ext
> running install
> /home/.../python3.7/site-packages/setuptools/command/install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
>   setuptools.SetuptoolsDeprecationWarning,
> /home/.../python3.7/site-packages/setuptools/command/easy_install.py:159: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
>   EasyInstallDeprecationWarning,
> running bdist_egg
> running egg_info
> writing manifest file 'MultiScaleDeformableAttention.egg-info/SOURCES.txt'
> running install_lib
> creating build/bdist.linux-x86_64/egg
> creating build/bdist.linux-x86_64/egg/functions
> creating build/bdist.linux-x86_64/egg/modules
> byte-compiling build/bdist.linux-x86_64/egg/functions/__init__.py to __init__.cpython-37.pyc
> byte-compiling build/bdist.linux-x86_64/egg/functions/ms_deform_attn_func.py to ms_deform_attn_func.cpython-37.pyc
> byte-compiling build/bdist.linux-x86_64/egg/modules/ms_deform_attn.py to ms_deform_attn.cpython-37.pyc
> byte-compiling build/bdist.linux-x86_64/egg/modules/__init__.py to __init__.cpython-37.pyc
> byte-compiling build/bdist.linux-x86_64/egg/MultiScaleDeformableAttention.py to MultiScaleDeformableAttention.cpython-37.pyc
> creating build/bdist.linux-x86_64/egg/EGG-INFO
> copying MultiScaleDeformableAttention.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
> copying MultiScaleDeformableAttention.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
> copying MultiScaleDeformableAttention.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
> copying MultiScaleDeformableAttention.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
> zip_safe flag not set; analyzing archive contents...
> __pycache__.MultiScaleDeformableAttention.cpython-37: module references __file__
> removing 'build/bdist.linux-x86_64/egg' (and everything under it)
> creating /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg
> Extracting MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg to /home/.../python3.7/site-packages
> byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/MultiScaleDeformableAttention.py to MultiScaleDeformableAttention.cpython-37.pyc
> byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/functions/__init__.py to __init__.cpython-37.pyc
> byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/functions/ms_deform_attn_func.py to ms_deform_attn_func.cpython-37.pyc
> byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/modules/__init__.py to __init__.cpython-37.pyc
> byte-compiling /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg/modules/ms_deform_attn.py to ms_deform_attn.cpython-37.pyc
> Adding MultiScaleDeformableAttention 1.0 to easy-install.pth file
> 
> Installed /home/.../python3.7/site-packages/MultiScaleDeformableAttention-1.0-py3.7-linux-x86_64.egg
> Processing dependencies for MultiScaleDeformableAttention==1.0
> Finished processing dependencies for MultiScaleDeformableAttention==1.0
> ```
> 
> Did this installation correctly finish?

have you solve this problem?

**Comment by ningerlove** [comment]
undefined symbol: _ZN3c104cuda20getCurrentCUDAStreamEa
have you solve this problem?

**Comment by ningerlove** [comment]
> undefined symbol: _ZN3c104cuda20getCurrentCUDAStreamEa have you solve this problem?
solved by pip install torch==1.8.0+cu111 torchvision==0.9.0+cu111 torchaudio==0.8.0 -f https://download.pytorch.org/whl/torch_stable.html

**Comment by ningerlove** [comment]
> pip install torch==1.8.0+cu111 torchvision==0.9.0+cu111 torchaudio==0.8.0 -f https://download.pytorch.org/whl/torch_stable.html
pip install torch==1.8.0+cu111 torchvision==0.9.0+cu111 torchaudio==0.8.0 -f https://download.pytorch.org/whl/torch_stable.html
